### PR TITLE
[FR] Add required-fields option to import-rules

### PR DESCRIPTION
**Pull request by Mikaayenson**
## Issues

None, but related to several PRs testing import rules features.

## Summary

While testing other PRs, I had to manually change this line just to fully test fields that are not required. 

- This exposes a required-only option to the import-rules CLI so that the default isn't set to not import required fields. 
- Updates the CLI.md
- ⚠️ NOTE: This is a breaking change which will prompt users for fields if the `--required-only` flag is not supplied.

## Testing

- If the CI passes, this should be g2g
- Test the import command with the new field:

<details><summary>Import-Rules CLI</summary>
<p>

```bash
(detection-rules-build) ➜  detection-rules git:(update_import_rules) ✗ python -m detection_rules import-rules /Users/stryker/Downloads/rules_export\(2\).ndjson
Loaded config file: /Users/stryker/workspace/ElasticGitHub/detection-rules/.detection-rules-cfg.json

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

[+] Building rule for /Users/stryker/workspace/ElasticGitHub/detection-rules/rules/exchange_mailbox_export_via_powershell.toml
actions (multi, comma separated): 
alert_suppression: 
building_block_type: 
data_view_id: 
exceptions_list (multi, comma separated): 
meta: 
risk_score_mapping (multi, comma separated): 
rule_name_override: 
setup: 
severity_mapping (multi, comma separated): 
add mitre tactic? [y/N]: 
throttle: 
timeline_id: 
timeline_title: 
(detection-rules-build) ➜  detection-rules git:(update_import_rules) ✗ python -m detection_rules import-rules /Users/stryker/Downloads/rules_export\(2\).ndjson --required-only

Loaded config file: /Users/stryker/workspace/ElasticGitHub/detection-rules/.detection-rules-cfg.json

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

[+] Building rule for /Users/stryker/workspace/ElasticGitHub/detection-rules/rules/exchange_mailbox_export_via_powershell.toml
(detection-rules-build) ➜  detection-rules git:(update_import_rules) ✗ 
```

</p>
</details>

### CODE REVIEW

File: detection_rules/main.py [hunk 1]
```diff
@@ -94,8 +94,9 @@ def generate_rules_index(ctx: click.Context, query, overwrite, save_files=True):
 
 @root.command('import-rules')
 @click.argument('input-file', type=click.Path(dir_okay=False, exists=True), nargs=-1, required=False)
+@click.option('--required-only', is_flag=True, help='Only prompt for required fields')
```
Comment (terrancedejesus) [inline detection_rules/main.py:97]: Should we add a default?

**Review by terrancedejesus — APPROVED** [review]
Makes sense to me. Would love to see import abstracted maybe in the future to be more standalone from the create rule prompt? Cheers 👍🏽

File: detection_rules/main.py (hunk 1 above)
Comment (eric-forte-elastic) [reply detection_rules/main.py:97]: Agreed. currently for me in testing, I think it defaults to `False` 

<details><summary>Test Run</summary>
<p>

```shell
detection-rules on  update_import_rules [$!?] is  v0.1.0 via  v3.12.2 (detection-rules-build) on  eric.forte
❯ python -m detection_rules import-rules /home/forteea1/Code/clean_mains/detection-rules/exports/20240314T190337L.ndjson

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

[+] Building rule for /home/forteea1/Code/clean_mains/detection-rules/rules/threat_intel_ip_address_indicator_match.toml
actions (multi, comma separated): test
alert_suppression:
building_block_type:
concurrent_searches:
data_view_id:
enabled:
exceptions_list (multi, comma separated):
false_positives (multi, comma separated):
filters (multi, comma separated):
items_per_search:
max_signals:
meta:
related_integrations (multi, comma separated):
risk_score_mapping (multi, comma separated):
rule_name_override:
severity_mapping (multi, comma separated):
add mitre tactic? [y/N]:
throttle:
to:


```

</p>
</details>

**Review by eric-forte-elastic — APPROVED** [review]
Manual review and testing looks good :green_circle: 

As noted above too, the default behavior I think is to not use this feature. Happy either way, but just to make sure this was intended.

<details><summary>Test Results with Default</summary>
<p>

```shell

<details><summary>Test Run</summary>
<p>

```shell
detection-rules on  update_import_rules [$!?] is  v0.1.0 via  v3.12.2 (detection-rules-build) on  eric.forte
❯ python -m detection_rules import-rules /home/forteea1/Code/clean_mains/detection-rules/exports/20240314T190337L.ndjson

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

[+] Building rule for /home/forteea1/Code/clean_mains/detection-rules/rules/threat_intel_ip_address_indicator_match.toml
actions (multi, comma separated): test
alert_suppression:
building_block_type:
concurrent_searches:
data_view_id:
enabled:
exceptions_list (multi, comma separated):
false_positives (multi, comma separated):
filters (multi, comma separated):
items_per_search:
max_signals:
meta:
related_integrations (multi, comma separated):
risk_score_mapping (multi, comma separated):
rule_name_override:
severity_mapping (multi, comma separated):
add mitre tactic? [y/N]:
throttle:
to:


```

</p>
</details> 

<details><summary>Test results with Required Only</summary>
<p>

```shell
detection-rules on  update_import_rules [$!?] is  v0.1.0 via  v3.12.2 (detection-rules-build) on  eric.forte took 11s
❯ python -m detection_rules import-rules /home/forteea1/Code/clean_mains/detection-rules/exports/20240314T190337L.ndjson --required-only

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

[+] Building rule for /home/forteea1/Code/clean_mains/detection-rules/rules/threat_intel_ip_address_indicator_match.toml


```

</p>
</details>

File: detection_rules/main.py (hunk 1 above)
Comment (Mikaayenson) [reply detection_rules/main.py:97]: I was thinking this but it is a flag. We could rename it to something that's like `import-all-fields`. I also went with this field name to be consistent with the call to `create-rule`. Thoughts?

**Comment by Mikaayenson** [comment]
> Makes sense to me. Would love to see import abstracted maybe in the future to be more standalone from the create rule prompt? Cheers 👍🏽

There is a `create-rule` CLI option, but I think you're referring to the `rule_prompt` method would will import all provided fields in the NDJSON and prompt for missing fields.  Not sure how we would abstract this even more but that rule_prompt method needs to be refactored altogether.

**Pull request merged**